src/creds/VerifiableCredential.js

```diff
@@ -290,6 +290,36 @@ async function cryptographicallySecureVerify(credential, verifyAttestationFunc,
   return true;
 }
 
+/**
+ * Verify if a credential was granted for a specific requester and requestId.
+ * @param credential - A credential object with expirationDate, claim and proof
+ * @return true if verified, false otherwise.
+ */
+function requesterGrantVerify(credential, requesterId, requestId, keyName) {
+  const label = _.get(credential.proof, 'anchor.subject.label');
+  const anchorPubKey = _.get(credential.proof, 'anchor.subject.pub');
+  const anchorData = _.get(credential.proof, 'anchor.subject.data');
+
+  if (_.isEmpty(credential.granted) || _.isEmpty(label) || _.isEmpty(anchorPubKey)) {
+    return false;
+  }
+
+  const stringToHash = `${label}${anchorData}${requesterId}${requestId}`;
+  const hexHash = sha256(stringToHash);
+
+  const cryptoManager = services.container.CryptoManager;
+
+  let verifyKey = keyName;
+  if (_.isEmpty(verifyKey)) {
+    if (!_.isFunction(cryptoManager.installKey)) {
+      throw new Error('CryptoManager does not support installKey, please use a `keyName` instead.');
+    }
+    verifyKey = `TEMP_KEY_NAME_${new Date().getTime()}`;
```

Comment:
Why this temp_key?

Reply:
When the keyName is not specified it creates a temporary key in the cryptoManager module from the public key in the credential (anchorPubKey).